community/en/tiny_imagenet_custom_resnet.ipynb

```diff
@@ -0,0 +1,552 @@
+    "        inputShape = (height, width, depth)\n",
+    "        chanDim = -1\n",
+    "\n",
+    "        inputs = tf.keras.Input(shape=inputShape)\n",
+    "        x = BatchNormalization(axis=chanDim, epsilon=bnEps, momentum=bnMom)(inputs)\n",
+    "\n",
+    "        x = Conv2D(filters[0], (5, 5), use_bias=False, padding=\"same\", kernel_regularizer=l2(reg))(x)\n",
+    "        x = BatchNormalization(axis=chanDim, epsilon=bnEps, momentum=bnMom)(x)\n",
+    "        x = Activation(\"relu\")(x)\n",
+    "        x = ZeroPadding2D((1, 1))(x)\n",
+    "        x = MaxPooling2D((3, 3), strides=(2, 2))(x)\n",
+    "    \n",
+    "        for i in range(0, len(stages)):\n",
+    "            stride = (1, 1) if i == 0 else (2, 2)\n",
+    "            x = ResNet.residual_block(x, filters[i + 1], stride, chanDim, red=True, bnEps=bnEps, bnMom=bnMom)\n",
+    "\n",
+    "            for j in range(0, stages[i] - 1):\n",
+    "                x = ResNet.residual_block(x, filters[i + 1], (1, 1), chanDim, bnEps=bnEps, bnMom=bnMom)\n",
+    "\n",
+    "        x = BatchNormalization(axis=chanDim, epsilon=bnEps, momentum=bnMom)(x)\n",
+    "        x = Activation(\"relu\")(x)\n",
+    "        x = AveragePooling2D((8, 8))(x)\n",
+    "  \n",
+    "        x = Conv2D(200, (1,1), kernel_regularizer=l2(reg))(x)\n",
+    "        x = Flatten()(x)\n",
+    "        x = Activation(\"softmax\")(x)\n",
+    "\n",
+    "        model = tf.keras.Model(inputs, x, name=\"resnet\")\n",
+    "\n",
+    "        return model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = ResNet.build(64, 64, 3, 200, (3, 4, 6), (64, 128, 256, 512), reg=0.0005)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tensorflow.keras.optimizers import Adam\n",
+    "model.compile(loss=\"categorical_crossentropy\", optimizer=opt, metrics=[\"accuracy\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using fit_generator to train the model\n",
+    "ImageDataGenerator is best suited for augmenting images on the fly and training the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.fit_generator(\n",
+    "  train_generator,\n",
+    "  steps_per_epoch=100000 // 64,\n",
+    "  validation_data=val_generator,\n",
+    "  validation_steps=10000 // 64,\n",
+    "  epochs=5,\n",
+    "  max_queue_size=64 * 2,\n",
+    "  callbacks=callbacks, verbose=1\n",
```

Comment:
?????

These were never defined.

Reply:
Yes, I have fixed it. It was supposed to be imported from a utils file.